Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/Editor/WindowsMixedRealityXRSDKConfigurationChecker.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System.Collections.Generic;
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.XRSDK.WindowsMixedReality
+{
+    /// <summary>
+    /// Class to perform checks for configuration checks for the WMR XR SDK provider.
+    /// </summary>
+    [InitializeOnLoad]
+    static class WindowsMixedRealityXRSDKConfigurationChecker
+    {
+        private const string AsmDefFileName = "Microsoft.MixedReality.Toolkit.Providers.XRSDK.WMR.asmdef";
+        private const string WindowsMixedRealityReference = "Unity.XR.WindowsMixedReality";
+
+#if UNITY_2019_3_OR_NEWER
+        private static readonly VersionDefine WindowsMixedRealityDefine = new VersionDefine("com.unity.xr.windowsmr", "", "WMR_ENABLED");
+#endif // UNITY_2019_3_OR_NEWER
+
+        static WindowsMixedRealityXRSDKConfigurationChecker()
+        {
+            UpdateAsmDef();
+        }
+
+        /// <summary>
+        /// Updates the assembly definition to contain the appropriate references based on the Unity version.
+        /// </summary>
+        /// <remarks>
+        /// Versions of Unity may have different factorings of components. To address this, the UpdateAsmDef
+        /// method conditionally compiles for the version currently in use.
+        /// To ensure proper compilation on each Unity version, the following steps are performed:
+        /// - Load the Microsoft.MixedReality.Toolkit.Providers.XRSDK.WMR.asmdef file
+        /// - If Unity 2018: nothing
+        /// - If Unity 2019 and newer: Unity.XR.WindowsMixedReality
+        /// - Save the Microsoft.MixedReality.Toolkit.Providers.XRSDK.WMR.asmdef file
+        /// This will result in Unity reloading the assembly with the appropriate dependencies.
+        /// </remarks>
+        private static void UpdateAsmDef()
+        {
+            FileInfo[] asmDefFiles = FileUtilities.FindFilesInAssets(AsmDefFileName);
```

Comment:
The lines 45-60ish are some of the stuff you were planning on refactoring on a follow up right? I noticed there are some similarities/reuse with the other config checker.

Reply:
Yep! Basically everything except the asmdef name and specific defines is duplicate logic.